### PR TITLE
Improve code coverage

### PR DESCRIPTION
**Pull request by alexbostock**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

I've added tests for a few commands to improve code coverage (issue #4763).

### CODE REVIEW

File: Library/Homebrew/test/cmd/commands_spec.rb [hunk 1]
```diff
@@ -6,6 +6,12 @@
   it "prints a list of all available commands" do
     expect { brew "commands" }
       .to output(/Built-in commands/).to_stdout
+      .and not_to_output.to_stderr
+      .and be_a_success
+
+    expect { brew "commands", "--quiet" }
```
Comment (reitermarkus) [inline Library/Homebrew/test/cmd/commands_spec.rb]: This should be a separate test and should check that the output doesn't include `Built-in commands`.

File: Library/Homebrew/test/cmd/info_spec.rb [hunk 2]
```diff
@@ -8,6 +8,11 @@
       .to output(/testball: stable 0.1/).to_stdout
       .and not_to_output.to_stderr
       .and be_a_success
+
+    expect { brew "info", "testball", "--json=v1" }
```
[the remark this answers is not in the document]
Comment (alexbostock) [reply Library/Homebrew/test/cmd/info_spec.rb]: I can't find an elegant way to parse the output (other than with regex), unless you think it should redirect stdout?

File: Library/Homebrew/test/cmd/commands_spec.rb [hunk 3]
```diff
@@ -6,8 +6,18 @@
...
       .to output(/Built-in commands/).to_stdout
+      .and not_to_output.to_stderr
       .and be_a_success
   end
+
+  it "prints a list without headers with the --quiet flag" do
+    expect { brew "commands", "--quiet" }
+      .to_not output(/Built-in commands/).to_stdout
+
+    expect { brew "commands", "--quiet" }
```
Comment (reitermarkus) [inline Library/Homebrew/test/cmd/commands_spec.rb]: Can you also connect these with `.and`?

File: Library/Homebrew/test/cmd/info_spec.rb [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
+  it "prints information about a given Formula" do
     expect { brew "info", "testball" }
       .to output(/testball: stable 0.1/).to_stdout
       .and not_to_output.to_stderr
       .and be_a_success
   end
+
+  it "prints as json with the --json=v1 flag" do
+    expect { brew "info", "testball", "--json=v1" }
+      .to output(/\{.+testball.+\}/).to_stdout
```
Comment (reitermarkus) [inline Library/Homebrew/test/cmd/info_spec.rb]: You can add a custom `a_json_string` matcher to check if this is parsable.

**Comment by reitermarkus** [comment]
Thanks, @alexbostock! Great work again. 👍

**Comment by MikeMcQuaid** [comment]
Sorry that I'm late to reviewing this. I just noticed this adds a bunch more integration tests which unfortunately isn't desirable as it slows down the test suite. My wording maybe wasn't clear in https://github.com/Homebrew/brew/issues/4763 but:

> This can be done by adding integration tests (only) for missing cmd and dev-cmd commands (e.g. https://codecov.io/gh/Homebrew/brew/src/master/Library/Homebrew/cmd/diy.rb)

What I was trying to say is that commands that already have integration tests (e.g. `brew info`) should prioritise unit tests rather than integration tests. This is because integration tests are pretty dramatically slower than unit tests.

Thoughts (including @reitermarkus)?

**Comment by alexbostock** [comment]
@MikeMcQuaid, sorry about that. Please could you clarify what you mean by integration tests and unit tests?

**Comment by MikeMcQuaid** [comment]
@alexbostock An integration test is one which has `describe "brew ...", :integration_test do` at the top of the file. These are executed by shelling out to a new Ruby process (which is slow). A unit test is any other one that is not an integration test (for the same of this discussion)

**Comment by alexbostock** [comment]
Thank you, @MikeMcQuaid. I'll bear that in mind in future. Should I go back and change what I've done in this PR?

**Comment by MikeMcQuaid** [comment]
@alexbostock That would be great, yes 🙇

**Pull request merged**